src/chem/Chain.js

```diff
@@ -36,6 +46,38 @@ class Chain {
     return this._residues;
   }
 
+  getType() {
+    return this._type;
+  }
+
+  _determineType() {
+    const residues = this._residues;
+
+    for (let i = 0, n = residues.length; i < n; ++i) {
+      if ((residues[i]._type.flags & ResidueType.Flags.NUCLEIC) !== 0) {
+        this._type = CHAIN_TYPE.NUCLEIC;
+        break;
+      } else if ((residues[i]._type.flags & ResidueType.Flags.PROTEIN) !== 0) {
+        this._type = CHAIN_TYPE.AMINO;
+        break;
+      }
```

Comment:
Should we rely on the value filled in the constructor or reset the type to UNKNOWN/GENERIC just before the loop?